src/solve.jl

```diff
@@ -113,8 +113,16 @@ function DiffEqBase.__init(
 
     # create separate copies u and uprev, not pointing to integrator.u or integrator.uprev,
     # to cache uprev with correct dimensions and types
-    u = recursivecopy(integrator.u)
-    uprev = recursivecopy(integrator.uprev)
+    if typeof(prob.u0) <: Tuple
+        u = ArrayPartition(prob.u0,Val{true})
```

Comment:
We should really be doing the conversion to ArrayPartitions inside of DiffEqBase instead, but that's a whole different story.